Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorContextMenu.cpp

```diff
@@ -0,0 +1,226 @@
+    QMenu::item:disabled:hover {
+      background-color: #FFFFFF;
+      color: #C8C8C8;
+    }
+
+    QMenu::right-arrow {
+      height: %7px;
+      image: url(:/Icons/arrow-right-black.svg);
+      padding-right: %6px;
+      width: %8px;
+    }
+
+    QMenu::right-arrow:selected {
+      height: %7px;
+      image: url(:/Icons/arrow-right-white.svg);
+      padding-right: %6px;
+      width: %8px;
+    }
+
+    QMenu::item:selected, QMenu::item:hover {
+      background-color: #8D78EC;
+      color: #FFFFFF;
+    })")
+    .arg(scale_height(1)).arg(scale_width(1))
+    .arg(scale_height(12)).arg(scale_height(5))
+    .arg(scale_height(3)).arg(scale_width(8))
+    .arg(scale_height(6)).arg(scale_width(4)));
+  m_table_columns_menu = new QMenu(tr("Table Columns"), this);
+  m_table_columns_menu->installEventFilter(this);
+  m_table_columns_menu->setFixedSize(scale(140, 150));
+  m_table_columns_menu->setContentsMargins(0, scale_height(0), 0,
+    scale_height(4));
+  addMenu(m_table_columns_menu);
+  auto security_action = new QWidgetAction(this);
+  add_check_box(tr("Security"), security_action);
+  connect(static_cast<QCheckBox*>(security_action->defaultWidget()),
+    &QCheckBox::stateChanged, [=] (int state) {
+        m_security_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(security_action);
+  auto side_action = new QWidgetAction(this);
+  add_check_box(tr("Side"), side_action);
+  connect(static_cast<QCheckBox*>(side_action->defaultWidget()),
+    &QCheckBox::stateChanged, [=] (int state) {
+        m_side_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(side_action);
+  auto size_action = new QWidgetAction(this);
+  add_check_box(tr("Size"), size_action);
+  connect(static_cast<QCheckBox*>(size_action->defaultWidget()),
+    &QCheckBox::stateChanged, [=] (int state) {
+        m_size_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(size_action);
+  auto ref_px_action = new QWidgetAction(this);
+  add_check_box(tr("Reference Px"), ref_px_action);
+  connect(static_cast<QCheckBox*>(ref_px_action->defaultWidget()),
+    &QCheckBox::stateChanged, [=] (int state) {
+        m_ref_px_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(ref_px_action);
+  auto date_action = new QWidgetAction(this);
+  add_check_box(tr("Date"), date_action);
+  connect(static_cast<QCheckBox*>(date_action->defaultWidget()),
+    &QCheckBox::stateChanged, [=] (int state) {
+        m_date_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(date_action);
+  auto time_action = new QWidgetAction(this);
+  add_check_box(tr("Time"), time_action);
+  connect(static_cast<QCheckBox*>(time_action->defaultWidget()),
+    &QCheckBox::stateChanged, [=] (int state) {
+        m_time_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(time_action);
+  auto notional_value_action = new QWidgetAction(this);
```

Comment:
This action connection routine is very similar for all actions. Check if you can factor it into a function taking the name and signal as arguments.

Reply:
This is a good idea.